src/runtime/auto-prompt-manager.js

```diff
@@ -190,17 +192,23 @@ export class AutoPromptManager {
     params
   ) {
     // Override autoPromptType if it is undefined.
-    params.autoPromptType =
-      params.autoPromptType ||
-      this.getAutoPromptType_(article?.audienceActions?.actions);
+    params.autoPromptType ??= this.getAutoPromptType_(
+      article?.audienceActions?.actions
+    );
 
+    // Override isClosable if isAccessibleForFree is defined.
+    const isClosable =
+      params.isAccessibleForFree !== undefined
+        ? params.isAccessibleForFree
+        : params.autoPromptType !== AutoPromptType.SUBSCRIPTION &&
+          params.autoPromptType !== AutoPromptType.SUBSCRIPTION_LARGE;
```

Comment:
nit: seems like there are lots of duplicate logic to define subscription/contribution based on `AutoPromptType`. Would it be useful to create a function like `isSubscriptionPrompt`?

Reply:
Added some helper functions.